test_programs/plonky2_prove_failure/div_by_0/src/main.nr

```diff
@@ -0,0 +1,3 @@
+fn main(x: u32, y: u32, z: u32) {
+    assert((x / y) == z);
+}
```

Comment:
This test is expected to fail and the output it produces is `Failed constraint`. One may assume that the error has happened because the `assert` has failed which is not the case. I suggest changing the code to
```
fn main(x: u32, y: u32) {
    let _ = x / y;
}
```
Which fails with the same error. We can add as a future task to fix the output of said error.

Reply:
No, the assert is *not* meant to fail. The assert is meant to exercise the div generator in the plonky2 backend, which:
1) shouldn't crash/panic, because of the division by 0
2) generates a result of 0: (5 / 0 == 0)
This test makes sure the program doesn't succeed, despite of 2).
Would changing this to `let _ = x / y;` test the same things? Wouldn't this just be optimized away by the compiler and excluded from the plonky2 circuit entirely, since the result is not used? Note: I haven't tested this, but I prefer the assert to stay, as it doesn't hurt anybody.

Reply:
Okay, it's good as it is.
Note: Yes, I have tested it with `let _ = x / y;`. The compiler doesn't optimize it away and it also gives the same error as the current assert test.